norfair/motion.py

```diff
@@ -0,0 +1,161 @@
+
+    try:
+        homography_matrix = homography_matrix @ data
+    except (TypeError, ValueError):
+        pass
+
+    inverse_homography_matrix = np.linalg.inv(homography_matrix)
+
+    if update_prvs:
+        data = homography_matrix
+
+    def abs_to_rel(points: np.array):
+        ones = np.ones((len(points), 1))
+        points_with_ones = np.hstack((points, ones))
+        points_transformed = points_with_ones @ homography_matrix.T
+        points_transformed = points_transformed / points_transformed[:, -1].reshape(
+            -1, 1
+        )
+        return points_transformed[:, :2]
+
+    def rel_to_abs(points: np.array):
+        ones = np.ones((len(points), 1))
+        points_with_ones = np.hstack((points, ones))
+        points_transformed = points_with_ones @ inverse_homography_matrix.T
+        points_transformed = points_transformed / points_transformed[:, -1].reshape(
+            -1, 1
+        )
+        return points_transformed[:, :2]
+
+    return update_prvs, data, abs_to_rel, rel_to_abs
+
+
+def get_sparse_flow(
+    gray_next, gray_prvs, prev_pts=None, max_points=300, min_distance=15, block_size=3
+):
+
+    if prev_pts is None:
+        # get points
+        prev_pts = cv2.goodFeaturesToTrack(
+            gray_prvs,
+            maxCorners=max_points,
+            qualityLevel=0.01,
+            minDistance=min_distance,
+            blockSize=block_size,
+        )
+
+    # compute optical flow
+    curr_pts, status, err = cv2.calcOpticalFlowPyrLK(
+        gray_prvs, gray_next, prev_pts, None
+    )
+    # filter valid points
+    idx = np.where(status == 1)[0]
+    prev_pts = prev_pts[idx].reshape((-1, 2))
+    curr_pts = curr_pts[idx].reshape((-1, 2))
+    return curr_pts, prev_pts
+
+
+class MotionEstimator:
+    def __init__(
+        self,
+        max_points=200,
+        min_distance=15,
+        block_size=3,
+        transformations_getter=get_homography_transformations,
+    ):
+        self.max_points = max_points
+        self.min_distance = min_distance
+        self.block_size = block_size
+
+        self.gray_prvs = None
+
+        self.transformations_getter = transformations_getter
+
+        self.data = (
+            None  # data that our transformation_getter needs to store and update
+        )
+        self.prev_pts = None
+
+    def update(self, frame, additional_arguments={}):
+        self.gray_next = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+        if self.gray_prvs is None:
+            self.gray_prvs = self.gray_next
+
+        curr_pts, self.prev_pts = get_sparse_flow(
+            self.gray_next,
+            self.gray_prvs,
+            self.prev_pts,
+            self.max_points,
+            self.min_distance,
+            self.block_size,
+        )
+
+        update_prvs, self.data, abs_to_rel, rel_to_abs = self.transformations_getter(
+            curr_pts, self.prev_pts, self.data, **additional_arguments
+        )
```

Comment:
I don't love the additional_argumets and the fact that the getter is storing data in this other class in order to function properly. 
How about making the getter a class, say `TransformationGetter` which accepts the additional_arguments in its init. The MotionEstimator can accept an instance of said class, call it and not handle the data.
For instance the translation would become:
``` python
class TransformationGetter:
    def __call__(self, curr_pts, prev_pts):
        raise NotImplementedError


class TranslationTransformationGetter(TransformationGetter):
    def __init__(self, bin_size=0.2, proportion_points_used_threshold=0.9) -> None:
        self.data = None
        self.bin_size = bin_size
        self.proportion_points_used_threshold = proportion_points_used_threshold

    def __call__(self, curr_pts, prev_pts):
        # code of get_translation_mode_transformations
        pass
```

And from the outside it can be used like this:

``` python
motion_estimator = MotionEstimator(
    max_points=200,
    min_distance=15,
    block_size=3,
    transformations_getter=TranslationTransformationGetter(bin_size=2),
)
motion_estimator.update(frame)
```